students/erica_edwards/Lesson03/mailroom.py

```diff
@@ -0,0 +1,97 @@
+
+from operator import itemgetter
+
+Donors = [('Sandy Pie', [75]),
+          ('Judy Smith', [75, 100, 1000]),
+          ('Mike Jones', [75, 1000]),
+          ('Joe Smith', [75, 100, 2000]),
+          ('Kelly Blue', [75, 150, 275])]
+
+def menu():
+    quit = False
+    while not quit:
+        print()
+        print('''Menu:   
+                    1. Send a Thank You.
+                
+                    2. Create a Report.
+            
+                    3. Quit.''')
+        print()
+        response = input("Please enter a number to make your selection. ")
+        if response == '1':
+            donor_name()
+            
+        if response == '2':
+            totals()    
+
+        if response == '3':
+            quit = True
+
+def list_donors():
+    print(Donors)
+    donor_name()
+
+def donor_name():
+    done = False
+    
+    # While collecting user input.
+    while not done:
+        donorName = input("Enter the full name of the donor. (or q to quit) ")
+        
+        if donorName == 'q':
+            return
+        # If user types list print Donors       
+        elif donorName.upper() == "LIST":
+            list_donors()
+            continue
+        else:  # user didn't type list    
+            if donorName not in [item[0] for item in Donors]:
+                Donors.append((donorName,list()))
+        donation(donorName)
+        done = True
+    
+def list_donors():
+    print(Donors)
+    donor_name()
+    return
+
+def donation(donorName):    
+    donationAmount = float(input("Enter the donation amount (or q to quit) "))
+    if donationAmount == 'q':
+        return
+    else:
+        donorIndex = [item[0] for item in Donors].index(donorName)
```

Comment:
same here, you're looping over full list and then list has to search index over the list - could be optimized with a single for loop or by passing the donor entry that you have already found prior